### PR TITLE
[WIP] Indexing exploration + embedbase >0.7.5 sync

### PR DESCRIPTION
**Pull request by louis030195**
# TLDR
- sync [Embedbase](https://github.com/different-ai/embedbase)@0.7.5
- added dev settings
- different indexing (exploration)
- option to ask embedbase to store the data (I want to query my index outside obsidian)

---

This is a bit overloaded PR, sorry about that, I think the indexing part should be improved before merge (and also we need to deploy https://github.com/different-ai/embedbase-ava with >embedbase@0.7.5 and switch to Supabase back-end at once)

image

I'm using this branch for my vault and using `https://embedbase-ava-dev-c6txy76x2q-uc.a.run.app` as Embedbase instance that is using Supabase

The indexing is trying to split markdown by headers and we embed a JSON directly

e.g.

note path: "Dogs.md"
note content: "# Dogs\n\nDogs are nice, why?\n- They evolved to be best human's friend!"

`{"content": "# Dogs\n\nDogs are nice, why?\n- They evolved to be best human's friend!", "path": "Dogs.md"}`

# TODOs before merging
- [ ] improve the indexing (use https://github.com/different-ai/embeddings-utils)
  - [ ] deal properly with large documents
- [ ] deploy https://github.com/different-ai/embedbase-ava on production with Supabase & (ping everyone that they have to index their vault again OR implement a painful migration from Pinecone)

# Trying this branch
- build the code and throw main.js in your vault
- use `https://embedbase-ava-dev-c6txy76x2q-uc.a.run.app` as embedbase api in settings

### CODE REVIEW

**Comment by louis030195** [comment]
PS: of course Ava does not store user data (only vectors) by default (now and then), aligned with Obsidian philosophy

**Comment by louis030195** [comment]
PS2: expect cold starts with embedbase instance (0 minimum instance)

**Comment by louis030195** [comment]
Asking questions to your vault using Siri + GPT4 + https://github.com/different-ai/embedbase

Screenshot 2023-03-17 at 17 25 57


Screenshot 2023-03-17 at 17 24 49


Screenshot 2023-03-17 at 17 25 11

Screenshot 2023-03-17 at 17 25 22
